pkg/binding/add.go

```diff
@@ -112,7 +113,12 @@ func (o *BindingClient) AddBindingToDevfile(
 		return obj, err
 	}
 
-	deploymentName := fmt.Sprintf("%s-app", obj.GetMetadataName())
+	componentName, err := component.GatherName(obj)
```

Comment:
I would prefer this function to be called during the context retrieval on genericclioptions.New(), and the component name stored in the context.

Reply:
Indeed, makes sense from the CLI perspective. I'll try to call this during the context retrieval. I see there is already a notion of component name in such context. However, I guess this would require passing this information down to the other functions/methods that might need it (instead of recomputing it from the Devfile object)..

Reply:
Yes, the value will need to be passed to business packages.

At some point, I would like the context to be built at the very beginning, and passed in the business client structures alongside the dependent clients, so we don't have to bother passing these values all around.

Reply:
I've changed this so as to call this function during context retrieval.
I'm leveraging the already existing `component` field in the Context (which, per my understanding, relied on the `--component` flag, but I am not sure this flag is still being used).
Please take a look at [718a6ad975b58f03f6b2fe5a99797a59e2975539](https://github.com/redhat-developer/odo/pull/6015/commits/718a6ad975b58f03f6b2fe5a99797a59e2975539) ([3f7ef4d58fd19805f033b34744caabdb58ba2e4f](https://github.com/redhat-developer/odo/pull/6015/commits/3f7ef4d58fd19805f033b34744caabdb58ba2e4f) amends that commit, following an issue I ran into when testing `odo list`).

Reply:
The CI failures should be fixed now. I had to rebase onto `main` to fix a few conflicts.